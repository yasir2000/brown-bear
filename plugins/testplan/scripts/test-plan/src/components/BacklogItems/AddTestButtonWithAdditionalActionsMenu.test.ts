/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { Wrapper } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import { createTestPlanLocalVue } from "../../helpers/local-vue-for-test";
import type { RootState } from "../../store/type";
import type { BacklogItem } from "../../type";
import { createStoreMock } from "../../../../../../../src/scripts/vue-components/store-wrapper-jest";
import AddTestButtonWithAdditionalActionsMenu from "./AddTestButtonWithAdditionalActionsMenu.vue";

describe("AddTestButtonWithAdditionalActionsMenu", () => {
    async function createWrapper(
        state: RootState,
        backlog_item: BacklogItem,
        should_empty_state_be_displayed: boolean
    ): Promise<Wrapper<AddTestButtonWithAdditionalActionsMenu>> {
        return shallowMount(AddTestButtonWithAdditionalActionsMenu, {
            localVue: await createTestPlanLocalVue(),
            propsData: {
                backlog_item,
                should_empty_state_be_displayed,
            },
            mocks: {
                $store: createStoreMock({
                    state,
                }),
            },
        });
    }

    it("Does not display the button if the backlog items says it can not add a test", async () => {
        const wrapper = await createWrapper(
            { milestone_id: 69, testdefinition_tracker_id: null } as RootState,
            { id: 123, can_add_a_test: false } as BacklogItem,
            false
        );

        expect(wrapper.element).toMatchInlineSnapshot(`<!---->`);
    });

    it("Does not display the button if there is no test definition tracker id", async () => {
        const wrapper = await createWrapper(
            { milestone_id: 69, testdefinition_tracker_id: null } as RootState,
            { id: 123, can_add_a_test: true } as BacklogItem,
            false
        );

        expect(wrapper.element).toMatchInlineSnapshot(`<!---->`);
    });

    it("Does not display the button if the test definitions are still loading", async () => {
        const wrapper = await createWrapper(
            { milestone_id: 69, testdefinition_tracker_id: 42 } as RootState,
            {
                id: 123,
                is_loading_test_definitions: true,
                can_add_a_test: true,
                is_expanded: true,
            } as BacklogItem,
            false
        );

        expect(wrapper.element).toMatchInlineSnapshot(`<!---->`);
    });

    it("Does not display the button if an error occurred during the load of the test definitions", async () => {
        const wrapper = await createWrapper(
            { milestone_id: 69, testdefinition_tracker_id: 42 } as RootState,
            {
                id: 123,
                is_loading_test_definitions: false,
                has_test_definitions_loading_error: true,
                can_add_a_test: true,
                is_expanded: true,
            } as BacklogItem,
            false
        );

        expect(wrapper.element).toMatchInlineSnapshot(`<!---->`);
    });

    it("Displays the button if conditions are ok", async () => {
        const wrapper = await createWrapper(
            { milestone_id: 69, testdefinition_tracker_id: 42 } as RootState,
            {
                id: 123,
                is_loading_test_definitions: false,
                has_test_definitions_loading_error: false,
                can_add_a_test: true,
                is_expanded: true,
            } as BacklogItem,
            false
        );

        expect(wrapper.element).toMatchSnapshot();
    });
});
