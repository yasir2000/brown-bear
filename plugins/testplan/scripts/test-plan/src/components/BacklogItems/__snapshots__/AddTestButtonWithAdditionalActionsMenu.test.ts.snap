// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTestButtonWithAdditionalActionsMenu Displays the button if conditions are ok 1`] = `
<div
  class="tlp-dropdown"
>
  <div
    class="tlp-dropdown-split-button test-plan-add-test-dropdown"
  >
    <a
      class="tlp-button-primary tlp-button-outline tlp-button-small tlp-dropdown-split-button-main"
      data-current-language="en_US"
      data-msgid="
            Create a new test
        "
      data-test="add-test-button"
      href="/plugins/tracker/?tracker=42&func=new-artifact&ttm_backlog_item_id=123&ttm_milestone_id=69"
    >
      
            Create a new test
        
    </a>
     
    <button
      class="tlp-button-primary tlp-button-outline tlp-append tlp-dropdown-split-button-caret tlp-button-small"
    >
      <i
        class="fa fa-caret-down"
      />
    </button>
  </div>
   
  <div
    class="tlp-dropdown-menu tlp-dropdown-menu-left"
    role="menu"
  >
    <a
      class="tlp-dropdown-menu-item"
      href="/plugins/tracker/?aid=123&ttm_backlog_item_id=123&ttm_milestone_id=69"
      role="menuitem"
    >
      <i
        class="fas fa-fw fa-pencil-alt"
      />
       
      <translate-stub
        tag="span"
        translateparams="[object Object]"
      >
        
                Edit %{ item_type } #%{ item_id }
            
      </translate-stub>
    </a>
  </div>
</div>
`;
